doc/languages-frameworks/maven.md

```diff
@@ -0,0 +1,191 @@
+---
+title: Maven
+author: Farid Zakaria
+date: 2020-10-15
+---
+
+# Maven
+
+Maven is a well-known build tool for the Java ecosystem however it has some challenges when integrating into the Nix build system.
+
+The following provides a list of common patterns with how to package a Maven project (or any JVM language that can export to Maven) as a Nix  package.
+
+## Double Invocation
+
+The double invocation is a _simple_ way to get around the problem that _nix-build_ may be sandboxed and have no Internet connectivity.
+
+It relies on the fact that Nix allows network access for **Fixed Output Derivations**.
+
+The first step will be to build the Maven project as a fixed-output derivation in order to collect the M2 repository.
+
+```nix
+{ stdenv, jdk11_headless, maven }:
+with stdenv;
+mkDerivation {
+    name = "maven-dependencies";
+    buildInputs = [ jdk11_headless maven ];
+    src = ./.;
+    buildPhase = ''
+      while mvn package -Dmaven.repo.local=$out/.m2 -Dmaven.wagon.rto=5000; [ $? = 1 ]; do
+        echo "timeout, restart maven to continue downloading"
+      done
+    '';
+    # keep only *.{pom,jar,sha1,nbm} and delete all ephemeral files with lastModified timestamps inside
+    installPhase = ''
+        find $out/.m2 -type f -regex '.+\\(\\.lastUpdated\\|resolver-status\\.properties\\|_remote\\.repositories\\)' -delete
+    '';
+    outputHashAlgo = "sha256";
+    outputHashMode = "recursive";
+    outputHash = "0000000000000000000000000000000000000000000000000000";
+  }
+```
+
+> It would be great if we could specify a remote repository as $HOME/.m2/repository to make downloads faster, however that functionality was removed. If you'd like add it to your settings.xml
+
+The build will fail, and tell you the expected hash to place. After which once you have, the build will return with a _/nix/store_ entry whose contents are the full Maven repository.
+
+> Some additional files are deleted that would cause the output hash to change potentially on subsequent runs.
+
+```bash
+❯ tree /nix/store/bxhr4p2x99jvqk027jsv250b861wklsq-dependencies/.m2
+├── backport-util-concurrent
+│   └── backport-util-concurrent
+│       └── 3.1
+│           ├── backport-util-concurrent-3.1.pom
+│           ├── backport-util-concurrent-3.1.pom.sha1
+│           └── _remote.repositories
+├── classworlds
+│   └── classworlds
+│       ├── 1.1
+│       │   ├── classworlds-1.1.jar
+│       │   ├── classworlds-1.1.jar.sha1
+│       │   ├── classworlds-1.1.pom
+│       │   ├── classworlds-1.1.pom.sha1
+│       │   └── _remote.repositories
+```
+
+If your package uses SNAPSHOT dependencies or version ranges; there is a strong likelihood that over-time your output hash will change.
+
+## buildMaven
+
+> ⚠️ Although _buildMaven_ is the "blessed" way within nixpkgs, it hasn't seen much activity in quite a while.
```

Comment:
If it picks up momentum this note may linger for quite a while. It'd be more helpful to tell the reader why they may not be able to use it and what could be done about it. On the other hand you've done your research, so putting a date in it also solves it afaic.

```suggestion
> ⚠️ Although _buildMaven_ is the "blessed" way within nixpkgs. As of 2020, it hasn't seen much activity in quite a while.
```